libraries/AP_TECS/AP_TECS.cpp

```diff
@@ -1436,3 +1446,12 @@ void AP_TECS::update_pitch_throttle(int32_t hgt_dem_cm,
     }
 #endif
 }
+
+bool AP_TECS::_in_takeoff_first_stage(void)
+{
+    return (_flight_stage == AP_FixedWing::FlightStage::TAKEOFF) && (
+        !_flags.reached_speed_takeoff || (
+            _tkoff_max_thr_alt > 0 && _height < _tkoff_max_thr_alt
```

Comment:
i think this may be a problem if we were doing a takeoff from down in a valley, as _height will be home relative, so this will be true during the whole takeoff
this conditional also needs to be broken apart to be easier to read, and commented

Reply:
I have removed this additional height condition and refactored the method with more comments.

For heights below `TKOFF_LVL_ALT` only the max throttle is now overwritten via the new `TECS::set_throttle_min()` method.